### PR TITLE
[modbus] itests: Fix testRefreshOnData

### LINKED ISSUE #9617: [modbus] Test ModbusDataHandlerTest#testRefreshOnData unstable
See https://ci.openhab.org/job/openHAB-Addons/230/

@ssalonen

```
TEST org.openhab.binding.modbus.tests.ModbusDataHandlerTest#testRefreshOnData() <<< ERROR: 
modbusPollerThingHandler.refresh();
Wanted 1 time:
-> at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.lambda$7(ModbusDataHandlerTest.java:805)
But was 10 times:
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
org.mockito.exceptions.verification.TooManyActualInvocations: 
modbusPollerThingHandler.refresh();
Wanted 1 time:
-> at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.lambda$7(ModbusDataHandlerTest.java:805)
But was 10 times:
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
-> at org.openhab.binding.modbus.internal.handler.ModbusDataThingHandler.lambda$0(ModbusDataThingHandler.java:174)
	at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.lambda$7(ModbusDataHandlerTest.java:805)
	at org.openhab.core.test.java.JavaTest.waitForAssert(JavaTest.java:152)
	at org.openhab.core.test.java.JavaTest.waitForAssert(JavaTest.java:120)
	at org.openhab.core.test.java.JavaTest.waitForAssert(JavaTest.java:84)
	at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.testRefreshOnData(ModbusDataHandlerTest.java:805)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:686)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:212)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:208)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at aQute.tester.bundle.engine.BundleDescriptor.executeChild(BundleDescriptor.java:49)
	at aQute.tester.bundle.engine.BundleEngine.lambda$executeBundle$7(BundleEngine.java:120)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at aQute.tester.bundle.engine.BundleEngine.executeBundle(BundleEngine.java:120)
	at aQute.tester.bundle.engine.BundleEngine.lambda$executeBundle$8(BundleEngine.java:133)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at aQute.tester.bundle.engine.BundleEngine.executeBundle(BundleEngine.java:133)
	at aQute.tester.bundle.engine.BundleEngine.lambda$execute$5(BundleEngine.java:100)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at aQute.tester.bundle.engine.BundleEngine.execute(BundleEngine.java:100)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at aQute.tester.junit.platform.Activator.test(Activator.java:352)
	at aQute.tester.junit.platform.Activator.automatic(Activator.java:317)
	at aQute.tester.junit.platform.Activator.run(Activator.java:217)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at aQute.launcher.Launcher.launch(Launcher.java:450)
	at aQute.launcher.Launcher.run(Launcher.java:179)
	at aQute.launcher.Launcher.main(Launcher.java:155)
	at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:170)
	at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:135)
	at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:52)
```

### PR DESCRIPTION
**Pull request by ssalonen**
Resolves #9617

Signed-off-by: Sami Salonen <ssalonen@gmail.com>

### CODE REVIEW

File: itests/org.openhab.binding.modbus.tests/src/main/java/org/openhab/binding/modbus/tests/ModbusDataHandlerTest.java [hunk 1]
```diff
@@ -798,8 +797,16 @@ public void testRefreshOnData() throws InterruptedException {
         assertThat(dataHandler.getThing().getStatus(), is(equalTo(ThingStatus.ONLINE)));
 
         verify(comms, never()).submitOneTimePoll(eq(request), notNull(), notNull());
-        // Reset initial REFRESH commands to data thing channels from the Core
+        ModbusPollerThingHandler handler = (ModbusPollerThingHandler) poller.getHandler();
+        // Wait for all channels to receive the REFRESH command (initiated by the core)
+        waitForAssert(
+                () -> verify((ModbusPollerThingHandler) poller.getHandler(), times(CHANNEL_TO_ACCEPTED_TYPE.size()))
+                        .refresh(),
+                2500, 50);
```
Comment (fwolter) [inline itests/org.openhab.binding.modbus.tests/src/main/java/org/openhab/binding/modbus/tests/ModbusDataHandlerTest.java:805]: Is there any reason not relying on the default timeout of `waitForAsser()`?

**Review by fwolter — CHANGES_REQUESTED** [review]
Thanks!

File: itests/org.openhab.binding.modbus.tests/src/main/java/org/openhab/binding/modbus/tests/ModbusDataHandlerTest.java (hunk 1 above)
Comment (ssalonen) [reply itests/org.openhab.binding.modbus.tests/src/main/java/org/openhab/binding/modbus/tests/ModbusDataHandlerTest.java:805]: Default should work just as fine I can imagine...

**Comment by ssalonen** [comment]
```
[ERROR] Failed to execute goal biz.aQute.bnd:bnd-testing-maven-plugin:5.1.2:testing (default) on project org.openhab.binding.astro.tests: 20 errors found -> [Help 1]
[ERROR] Failed to execute goal biz.aQute.bnd:bnd-testing-maven-plugin:5.1.2:testing (default) on project org.openhab.binding.avmfritz.tests: 18 errors found -> [Help 1]
```

which leads to

```
[INFO] openHAB Add-ons :: Integration Tests :: Modbus Binding Tests SKIPPED
```

**Review by fwolter — APPROVED** [review]

**Pull request merged**